rust/processor/src/models/default_models/transactions.rs

```diff
@@ -186,13 +194,35 @@ impl Transaction {
             .timestamp
             .as_ref()
             .expect("Transaction timestamp doesn't exist!");
+
+        let (wsc, wsc_detail) = WriteSetChangeModel::from_write_set_changes(
+            &transaction_info.changes,
+            version,
+            block_height,
+            deprecated_tables,
+        );
+
+        if deprecated_tables.contains(TableFlags::TRANSACTIONS) {
```

Comment:
It might be cleaner to do this inside of the models themselves, e.g. BlockMetadataTransaction::from_transaction. 

Maybe add a todo and we can fix later? This is a nit.